core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -785,7 +785,7 @@ class ReplicaManager(val config: KafkaConfig,
                                origin: AppendOrigin,
                                entriesPerPartition: Map[TopicPartition, MemoryRecords],
                                requiredAcks: Short): Map[TopicPartition, LogAppendResult] = {
-
+    val isTraceEnabled = logger.underlying.isTraceEnabled
```

Comment:
I need a way to do this more cleanly. Is it OK to put an isTraceEnabled call into the logger rather than access the underlying directly?